mypy/join.py

```diff
@@ -187,6 +187,11 @@ def visit_callable_type(self, t: CallableType) -> ProperType:
             if is_equivalent(t, self.s):
                 return combine_similar_callables(t, self.s)
             result = join_similar_callables(t, self.s)
+            # unless some or all items represent abstract objects,
+            # we set the flag: from_type_type to True
```

Comment:
This and below comment are not very helpful, because they are just repeating what happens in code. Maybe instead explain the _motivation_. Something like: "We set the from_type_type flag to suppress error when a collection of concrete class objects gets inferred as their common abstract superclass."

The same applies below. Also please start comments with an uppercase letter.